src/main/webapp/app/shared/components/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'tiny-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.css']
+})
+export class CheckboxComponent {
+  @Input() name: string;
+  @Input() click: string;
+
+  pushChanges(value: string): void {
+    this.onChange();
+  }
+
+  protected onChange = () => {};
```

Comment:
what is this lambda function for?

Reply:
Good catch 👍 

In this context, it's not useful. But I wanted to demonstrate that we can push changes or listen to the events from a shared component. As an example, if we have a shared textbox component we can update the value from the input box. Then we can listen to that on the onChange method with $event.target.value. These steps to be added to the shared `(click)="pushChanges('')"` in previous comment.